client.go

```diff
@@ -25,44 +25,52 @@ var clientServerConnection net.Conn
 var clientMasterConnection rpc.Server
 //Need a map from clientId->serverConnPort
 
+// client LogicalClock
+var clientLogicalClock = 0
 
 type ClientMaster int
 
 type PutArgs struct {
 	key, value string
-	clientId, serverId int
+	clientId, serverId, clientLogicalClock int
 }
 
 type GetArgs struct {
 	key string
 }
 
+type history struct {
+	action string
+	key, value string
+	clientId, serverId int
+	timeStamp map[int]int
+}
+
+var clientHistory []history
+
 var clientWaitGroup sync.WaitGroup
 
 func (t *ClientMaster) clientPut(clientId int, key string, value string) error {
+	// Increment clients logical clock on receiving a put request from the master
+	clientLogicalClock += 1
 
-	/*
-	get serverId from clientId from map in master
-	get serverBasePort from serverId from map in master
-	*/
-
+	// Make a put request to the connected server
 	if serverConn == 0 {
 		return errors.New("connection does not exist")
 	}
-
- 	args := &PutArgs{key, value, clientId, serverConn}
- 	var reply int
-
+ 	args := &PutArgs{key, value, clientId, serverConn, clientLogicalClock}
+ 	// reply contains vectorTimeStamp corresponding to this transaction
+ 	var reply shared.Clock
 	client := rpc.NewClient(clientServerConnection)
```

Comment:
Open a client for the first time and reuse it thereon instead of creating a client for every request. Look at getServerRpcClient function in server.go